src/cuda/CUDACore/allocate_device.cc

```diff
@@ -22,17 +22,25 @@ namespace cms::cuda {
       cudaCheck(allocator::getCachingDeviceAllocator().DeviceAllocate(dev, &ptr, nbytes, stream));
     } else {
       ScopedSetDevice setDeviceForThisScope(dev);
+#if CUDA_VERSION >= 11020
+      cudaCheck(cudaMallocAsync(&ptr, nbytes, stream));
+#else
```

Comment:
I would like to be able to force `cudaMalloc()` also on CUDA >= 11.2. How about an explicit macro, e.g. something along `CUDA_DISABLE_MALLOC_ASYNC` (possibly combined with the version check if the macro would disable the "async" version)?

Reply:
I've added `CUDA_DISABLE_ASYNC_ALLOCATOR` to follow the same naming scheme as `CUDA_DISABLE_CACHING_ALLOCATOR`